### PR TITLE
Add telemetry to track usage of generated code

### LINKED ISSUE #637: Log additional telemetry to track "% of cody code inserted from chats/commands"
Per request from this [PRD](https://docs.google.com/document/d/1RO6BiFILyciMy8UtkznX6Z4ewqzJhwmhg7CEKI-YrSg/edit) (and to get to parity with logging in VSCode), we need to start logging the events + metadata required to measure “% of cody code inserted or pasted from chats and commands”. Requirements below:

**Code generation events**  
These events log when Cody returns code via a chat or command/fixup response.

- Event: CodyJetBrainsPlugin:fixupResponse:hasCode
    - Behavior: fires when a fixup response from cody contains code
    - Metadata: logs the lineCount, the number of lines of code generated in the response
- Event: CodyJetBrainsPlugin:chatResponse:hasCode
    - Behavior: fires when a chat response from cody contains code
    - Metadata: logs the lineCount, the number of lines of code generated in the response

**Code insert events**
These events log when the user copies/inserts code returned by cody via a chat or command response. 

- Event: CodyJetBrainsPlugin:keyDown:Paste:clicked
    - Behavior: Fires on paste event from hot-keys (e.g. cmd+v)
    - Metadata: logs the lineCount, the number of lines of code generated in the response
- Event: CodyJetBrainsPlugin:insertButton:clicked
    - Behavior: fires when user clicks paste button in the extension
    - Metadata: logs the lineCount, the number of lines of code generated in the response and logs the lineCount metadata
- Event: CodyJetBrainsPlugin:inlineChat:Paste:clicked
    - Behavior: fires when user clicks paste button inline and logs the lineCount metadata
    - Metadata: logs the lineCount, the number of lines of code generated in the response and logs the lineCount metadata
 
Note that these events need to be instrumented as V1 to meet the customer's timeline requirements. 

cc @abeatrix who instrumented most (all?) of these events in VSCode - please let me know if I'm off-base on the behavior of any events!

cc @aramaraju and @taylorsperry FYI

### PR DESCRIPTION
**Pull request by pkukielka**
## Changes

Closes https://github.com/sourcegraph/jetbrains/issues/637

New code generation tracking events were added:

+  `CodyJetBrainsPlugin:chatResponse:hasCode`
+ `CodyJetBrainsPlugin:insertButton:clicked`
+ `CodyJetBrainsPlugin:copyButton:clicked`
+  `CodyJetBrainsPlugin:keyDown:Paste:clicked`

## Test plan

* Manual verification in debugger that proper events are set
* Check if events are present in GraphQL DB

### CODE REVIEW

File: src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java [hunk 1]
```diff
@@ -173,6 +180,17 @@ private static ActionListener getSendActionListener(EditorEx editor, JButton cop
           new Timer((int) Duration.ofSeconds(2).toMillis(), it -> copyButton.setText("Copy"));
       timer.setRepeats(false);
       timer.start();
+
+      Optional<@NotNull Project> project =
+          Arrays.stream(ProjectManager.getInstance().getOpenProjects()).findFirst();
+      if (project.isEmpty()) return;
```
Comment (mkondratek) [inline src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java:186]: maybe there is no reason for `copyCodeListener` to be static and we can just use `CodeEditorFactory::project`?

File: src/main/kotlin/com/sourcegraph/cody/chat/ui/SingleMessagePanel.kt [hunk 2]
```diff
@@ -108,6 +110,10 @@ class SingleMessagePanel(
         val listener = AttributionListener.UiThreadDecorator(lastPart.attribution)
         AttributionSearchCommand(project).onSnippetFinished(lastPart.text, sessionId, listener)
       }
+
+      ApplicationManager.getApplication().executeOnPooledThread {
+        GraphQlLogger.logCodeGenerationEvent(project, "chatResponse", "hasCode", lastPart.text)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/chat/ui/SingleMessagePanel.kt:115]: `lastPart.text` is the code, right?

File: src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java [hunk 3]
```diff
@@ -42,6 +47,8 @@ public class CodeEditorFactory {
   private final int rightMargin;
   public static final int spaceBetweenButtons = 5;
 
+  public static volatile String lastCopiedText = null;
```
Comment (mkondratek) [inline src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java:50]: oh, this is interesting... 🤔 so we store the copied text so we can know later if the pasted text is the copied one

Comment (mkondratek) [reply src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java:50]: a stupid question: does it work? 😅

**Review by mkondratek — APPROVED** [review]
lgtm

File: src/main/kotlin/com/sourcegraph/cody/chat/ui/SingleMessagePanel.kt (hunk 2 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/chat/ui/SingleMessagePanel.kt:115]: Yes

File: src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java (hunk 3 above)
Comment (pkukielka) [reply src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java:50]: It does, I checked with the debugger ;)

File: src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java (hunk 1 above)
Comment (pkukielka) [reply src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java:186]: Good catch, done.

**Pull request merged**